aws_lambda_powertools/logging/logger.py

```diff
@@ -118,7 +118,7 @@ def __init__(
     ):
         self.service = service or os.getenv("POWERTOOLS_SERVICE_NAME") or "service_undefined"
         self.sampling_rate = sampling_rate or os.getenv("POWERTOOLS_LOGGER_SAMPLE_RATE") or 0.0
-        self.log_level = level or os.getenv("LOG_LEVEL") or logging.INFO
+        self.log_level = level or os.getenv("LOG_LEVEL".upper()) or logging.INFO
```

Comment:
Not sure what is the purpose of  `.upper()` in here since `"LOG_LEVEL"` is explicitly set as all uppercase

Reply:
`LOG_LEVEL` is the env variable, and one particular customer ended up using "info" as the value for the env var - This solves that. 

In fact, I'm gonna turn this into a private method to cover the explicit `level` too

Reply:
Gotcha, your explanation makes sense, then you need the .upper outside of the getenv(LOG_LEVEL).UPPER()